src/Libraries/Microsoft.Extensions.AI.AzureAIInference/AzureAIInferenceChatClient.cs

```diff
@@ -490,5 +498,6 @@ private static List<ChatMessageContentItem> GetContentParts(IList<AIContent> con
 
     private static FunctionCallContent ParseCallContentFromJsonString(string json, string callId, string name) =>
         FunctionCallContent.CreateFromParsedArguments(json, callId, name,
-            argumentParser: static json => JsonSerializer.Deserialize(json, JsonContext.Default.IDictionaryStringObject)!);
+            argumentParser: static json => JsonSerializer.Deserialize(json,
+                (JsonTypeInfo<IDictionary<string, object>>)AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(IDictionary<string, object>)))!);
```

Comment:
We don't want to use ToolCallJsonSerializerOptions here as well?

Reply:
We could, but the callback delegate would no longer be static. Deserializing `object` is much more straightforward compared to serializing; it just uses `JsonElement` as the runtime type and so the default works for all intents and purposes.